File: biglist_filter_kentekens.py
#!/usr/bin/env python
from __future__ import print_function

import time
import csv

import sys  
import os
import time

start_time = time.time()

def main():  
	inputFile = os.path.realpath("./Open_Data_RDW__Gekentekende_voertuigen.csv")

	if not os.path.isfile(inputFile):
		print("File path={} does not exist.".format(inputFile))
		sys.exit()

	with open(inputFile) as fp:
		outputFiles = []
		for cnt, line in enumerate(fp):
			row = csv.reader([line]).next()
			if cnt == 0:
				index = ""
				for n, col in enumerate(row):
					if n == 0:
						index = col
					outputFiles.append(openOutputFile(n))
					outputFiles[n].write(str(index)+','+str(col)+"\n")
			else:
				index = ""
				for n, col in enumerate(row):
					if n == 0:
						index = col
					if col == '' or col == 'N.v.t.':
						continue
					outputFiles[n].write(str(index)+','+str(col)+"\n")
			# print("Line {}: {}".format(cnt, list(row)))
			if cnt % 100 == 0:
				printUpdate(fp)
			if cnt > 100000:
				break

		for file in outputFiles:
			file.close()
	print('')
def printUpdate(fp):
	print("\rPosition: {} Time: {} Remaining: {} Total Expected: {}".format( round(percentageOfFile(fp),3), formatTime(timeElapsed()), formatTime(timeRemaining(fp)), totalTime(fp)), end='')

def formatTime(t):
	return time.strftime('%H:%M:%S', time.gmtime(t))

def timeElapsed():
	return time.time() - start_time

def timeRemaining(fp):
	percent = percentageOfFile(fp)
	return timeElapsed() * (100 - percent) / percent

def totalTime(fp):
	return formatTime(timeElapsed() + timeRemaining(fp))

def percentageOfFile(fp):
	return float(fp.tell()) / float(os.fstat(fp.fileno()).st_size) * 100

def openOutputFile(number):
	return open(os.path.realpath('./Biglist_Outputs/'+str(number)+'.csv'), 'wb')

if __name__ == '__main__':  
	main()

File: biglist_filter_kentekens.rb
#!/usr/bin/env ruby

require 'csv'

$StartTime = Time.now

$files = []
def openOutputFiles(row)
	File.open('./Biglist_Outputs/index.csv', 'w') do |indexFile|
		row.each_with_index do |field, index|
			indexFile.puts(row[0] + ',' + field)
			$files << File.new('./Biglist_Outputs/' + index.to_s + '.csv', 'w')
			$files[index].puts(row[0] + ',' + field)
		end
	end
end

def writeOutput(row)
	row.each_with_index do |field, index|
		if field === nil
			next
		end
		if field === ''
			next
		end
		$files[index].puts(row[0] + ',' + field)
	end
end

def percentageOfFile(linenum)
	(linenum * 100 / 13500000.0)
end

def elapsed()
	Time.now - $StartTime
end

def remaining(linenum)
	elapsed * (100 - percentageOfFile(linenum)) / percentageOfFile(linenum)
end

def formatTime(total_time)
	[total_time / 3600, total_time / 60 % 60, total_time % 60].map { |t| t.to_i.to_s.rjust(2,'0') }.join(':')
end

def totalTime(linenum)
	formatTime(elapsed() + remaining(linenum))
end


rownum = 0

CSV.foreach('Open_Data_RDW__Gekentekende_voertuigen.csv') do |row|
	if rownum == 0
		openOutputFiles row
	else
		writeOutput row #row.inspect
	end
	rownum += 1
	if rownum % 10 == 0
		print "\r"
		print sprintf('%.3f', percentageOfFile(rownum).round(3))
		print '% Elapsed:'
		print formatTime elapsed
		print ' Remaining:'
		print formatTime remaining rownum
		print ' Total Expected:'
		print totalTime rownum

		# puts
		# break
	end
	if rownum > 100000
		break
	end
end

$files.each do |file|
	file.close
end

print "\n"